csharp/src/IceSSL/Endpoint.cs

```diff
@@ -209,8 +209,8 @@ internal sealed class EndpointFactoryI : IceInternal.EndpointFactoryWithUnderlyi
         public EndpointFactoryI(Instance instance, short type) : base(instance, type) => _instance = instance;
 
         public override IceInternal.IEndpointFactory
-        CloneWithUnderlying(IceInternal.ProtocolInstance inst, short underlying) =>
-            new EndpointFactoryI(new Instance(_instance.Engine(), inst.Type, inst.Protocol), underlying);
+        CloneWithUnderlying(IceInternal.TransportInstance inst, short underlying) =>
+            new EndpointFactoryI(new Instance(_instance.Engine(), inst.Type, inst.Transport), underlying);
 
```

Comment:
I know it is unrelated to the PR, but this code can improve by renaming things, here there is new Instance, _instance and inst